### ISSUE TITLE
How to map nested content in umbraco version 8? Exception thrown

### ISSUE THREAD

**Issue by rasolo** [AndyButland/UmbracoMapper#23]
Hi,

In the readme it says "Mapping of a collection of IPublishedContent (or, for v8, a collection of nested content IPublishedElement) items - e.g. from a node query picker - to a custom view model's collection.".

In version 7 I can create an `IEnumerable<IPublishedContent> `property with the same name (or use the "[PropertyMapping(SourceProperty =.." attribute" and umbracomapper will automatically map the property.

I try to do the same here in version 8 on my view model but with IEnumerable<IPublishedElement> and I get an exception:

  ```
Method Zone.UmbracoMapper.V8.UmbracoMapper.MapCollectionOfIPublishedElement: type argument
 'Umbraco.Core.Models.PublishedContent.IPublishedElement' violates the constraint of type parameter 'T'.
Description: An unhandled exception occurred during the execution of the current web request.
 Please review the stack trace for more information about the error and where it originated in the code.

Exception Details: System.Security.VerificationException: Method 

Zone.UmbracoMapper.V8.UmbracoMapper.MapCollectionOfIPublishedElement: type 
argument 'Umbraco.Core.Models.PublishedContent.IPublishedElement' 
violates the constraint of type parameter 'T'.

Source Error:


Line 32: 		{
Line 33: 			var model = new TModel();
Line 34: 			this.umbracoMapper.Map(content, model);
Line 35: 			this.PopulateModel(content, model);
Line 36: 			return model;
```

the variable content is the IPublishedContent content you get in the controller.

What I can do however is this:
```
var propertyToMapTo = new List<MyProperty>();
var propertyToMapToAsIPublishedElement = content.Value<IEnumerable<IPublishedElement>>("thePropertyAlias");

umbracoMapper.MapCollection(propertyToMapToAsIPublishedElement, propertyToMapTo);

viewModel.MyProperty = propertyToMapTo;
```
where I retrieve the IPublishedElement myself from the IPublishedContent you get in the controller.

But in version 7 I don't have to retrieve it, it gets automatically mapped using umbracomapper:
```
 var propertyToMapTo = new List<MyProperty>();
umbracoMapper.MapCollection(propertyToMapToAsIPublishedElement, content.ThePropertyInViewModel);

viewModel.MyProperty = propertyToMapTo;
```

Am I doing something wrong or?

I saw this issue: https://github.com/AndyButland/UmbracoMapper/issues/18
and I am running version 4.2.2

**Comment by AndyButland** [comment]
Not sure I've quite followed here so possibly I'm missing the issue being raised, but with regard the nested content question, mapping should work OK (either explicitly or via the "automapping").

Simplifying the example I've got in my tests, I've got a view model like this:

```
public class MyViewModel
{
    public IList<NestedItem> NestedItems { get; set; } = new List<NestedItem>();
}

public class NestedItem
{
	public string Text { get; set; }
}
```

Then with a content item containing a property of type "Nested Content" with an alias of "nestedContent", based on a document type containing a text string property with an alias of `text`.

```
var model = new MyViewModel();
Mapper.Map(CurrentPage, model);
```

Serves to populate the view model `NestedItems` collection with a collection drawn from the nested content items.
